polaris-react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -206,6 +212,11 @@ export function Tooltip({
         handleFocus();
       }}
       onBlur={() => {
+        if (hoverDelayTimeout.current) {
+          clearTimeout(hoverDelayTimeout.current);
+          hoverDelayTimeout.current = null;
+        }
```

Comment:
Prevents the `Tooltip` from flickering/reopening when clicking the activator before the `hoverDelayTimeout`

https://github.com/Shopify/polaris/assets/32409546/497a4b99-e8ee-4330-83ab-de632d36e897